app/models/phone_number_authentication.rb

```diff
@@ -0,0 +1,18 @@
+class PhoneNumberAuthentication < ApplicationRecord
+  has_secure_password
+
+  has_one :user_authentication, as: :authenticatable
+  has_one :master_user, through: :user_authentication
+
+  belongs_to :facility, foreign_key: 'registration_facility_id'
+
+  validates :phone_number, presence: true, uniqueness: true, case_sensitive: false
+  validates :password, allow_blank: true, length: { is: 4 }, format: { with: /[0-9]/, message: 'only allows numbers' }
```

Comment:
Is the password in this case the OTP? If so, currently it is 6 digits long IIRC. Is this length changing to 4 now?